cluster-autoscaler/cloudprovider/juju/juju_cloud_provider.go

```diff
@@ -200,3 +209,18 @@ func parseNodeGroupName(name string) (string, string, string, error) {
 	application := s[2]
 	return controller, model, application, nil
 }
+
+func readCloudConfigYaml(configRC io.ReadCloser) (jujuCloudConfig, error) {
+	t := jujuCloudConfig{}
+	b, err := io.ReadAll(configRC)
+	if err != nil {
+		return t, err
+	}
+
+	err = yaml.Unmarshal(b, &t)
+	if err != nil {
+		return t, err
+	}
+
+	return t, err
```

Comment:
there MAY be a go-land reason for this i don't understand, but looks like these are the same

```go
return t, err
```